nucypher/network/nodes.py

```diff
@@ -260,6 +263,50 @@ def abridged_node_details(node):
                 }
 
 
+class NodeSprout(PartiallyKwargifiedBytes):
+    """
+    An abridged node class designed for optimization of instantiation of > 100 nodes simultaneously.
+    """
+    verified_node = False
+
+    def __init__(self, node_metadata):
+        super().__init__(node_metadata)
+        self.checksum_address = to_checksum_address(node_metadata['public_address'][0])
+        self.nickname = nickname_from_seed(self.checksum_address)[0]
+        self.timestamp = maya.MayaDT(int.from_bytes(node_metadata['timestamp'][0], byteorder="big"))
+        self._hash = int.from_bytes(bytes(node_metadata['verifying_key'][0]), byteorder="big")
+
+    def __hash__(self):
+        return self._hash
+
+    def __repr__(self):
+        r = f"({self.__class__.__name__})⇀{self.nickname}↽ ({self.checksum_address})"
+        return r
+
+    def __bytes__(self):
+        b = super().__bytes__()
+
+        # We assume that the TEACHER_VERSION of this codebase is the version for this NodeSprout.
+        # This is probably true, right?  Might need to be re-examined someday if we have
+        # different node types of different versions.
+        version = Teacher.TEACHER_VERSION.to_bytes(2, "big")
+        return version + b
+
+    @property
+    def stamp(self) -> bytes:
+        return self.processed_objects['verifying_key'][0]
+
+    def mature(self):
+        mature_node = self.finish()
+
+        # As long as we're doing egregious workarounds, here's another one.  # TODO: 1481
+        filepath = mature_node._cert_store_function(certificate=mature_node.certificate)
+        mature_node.certificate_filepath = filepath
+
+        self.__class__ = mature_node.__class__
```

Comment:
Wow, serious magic!

Reply:
Too much for you?

I mean, I know that polymorphism and meta programming are often lumped in with more magic techniques, but I think there's something elegant about changing the class of the object in place.

Reply:
This was called "evolution" in Pokemon
![image](https://user-images.githubusercontent.com/2564234/71920102-5b939a00-3186-11ea-8ac6-20b68f8c83a7.png)